packages/react-form-state/src/FormState.ts

```diff
@@ -346,25 +336,27 @@ export default class FormState<
     return errors;
   }
 
-  private updateErrors(newErrors: FormError[]) {
+  private updateRemoteErrors(errors: RemoteError[]) {
     this.setState(({fields}: State<Fields>) => {
-      const newFields = {...(fields as any)};
-
-      const errors: ClientError[] = newErrors.map(({message, field}) => {
-        if (field == null) {
-          return {message};
-        }
+      const errorDictionary = errors.reduce(
+        (accumulator: any, {field, message}) => {
+          if (field == null) {
+            return accumulator;
+          }
 
-        if (isArray(field)) {
-          return {message, field: field.join('.')};
-        }
-
-        return {message, field};
-      });
+          return set(accumulator, field, message);
```

Comment:
You are using `set` from lodash because the `field` can be a path?

Reply:
yep it can be an array of path segments (that's what it looks like when they come from graphql)